src/profile-logic/profile-data.js

```diff
@@ -639,19 +639,48 @@ export function getTimingsForCallNodeIndex(
   return { forPath: pathTimings, forFunc: funcTimings, rootTime };
 }
 
+// This function computes the time range for a thread, using both its samples
+// and markers data.
 export function getTimeRangeForThread(
-  thread: Thread,
+  { samples, markers }: Thread,
   interval: Milliseconds
 ): StartEndRange {
-  if (thread.samples.length === 0) {
-    return { start: Infinity, end: -Infinity };
+  const result = { start: Infinity, end: -Infinity };
+
+  if (samples.length) {
```

Comment:
Optional: My preference in JavaScript is to always be explicit and not rely on the tricky coercion rules, e.g. `samples.length === 0`.